straxen/plugins/event_processing.py

```diff
@@ -728,35 +748,67 @@ def compute(self, events):
             s2_positions = np.vstack([events[f'{peak_type}s2_x'], events[f'{peak_type}s2_y']]).T
-
+            result[f"{peak_type}cs2_wo_elifecorr"] = np.zeros(len(result['time']))
+            result[f"{peak_type}cs2_area_fraction_top"] = np.zeros(len(result['time']))
+            result[f"{peak_type}cs2_wo_timecorr"] = np.zeros(len(result['time']))
+            result[f"{peak_type}cs2"] = np.zeros(len(result['time']))
+            result[f"{peak_type}cs2_bottom"] = np.zeros(len(result['time']))
+            
             # corrected s2 with s2 xy map only, i.e. no elife correction
             # this is for s2-only events which don't have drift time info
-            cs2_top_xycorr = (events[f'{peak_type}s2_area'] * events[f'{peak_type}s2_area_fraction_top'] /
-                                    self.s2_xy_map(s2_positions, map_name=s2_top_map_name))
-            cs2_bottom_xycorr = (events[f'{peak_type}s2_area'] *
-                                       (1 - events[f'{peak_type}s2_area_fraction_top']) /
-                                       self.s2_xy_map(s2_positions, map_name=s2_bottom_map_name))
-
-            # Correct for SEgain and extraction efficiency
-            seg_ee_corr = (self.se_gain / self.avg_se_gain) * self.rel_extraction_eff
-            cs2_top_wo_elifecorr = cs2_top_xycorr / seg_ee_corr
-            cs2_bottom_wo_elifecorr = cs2_bottom_xycorr / seg_ee_corr
-            result[f"{peak_type}cs2_wo_elifecorr"] = cs2_top_wo_elifecorr + cs2_bottom_wo_elifecorr
-
-            # cs2aft doesn't need elife/time corrections as they cancel
-            result[f"{peak_type}cs2_area_fraction_top"] = cs2_top_wo_elifecorr / result[f"{peak_type}cs2_wo_elifecorr"]
-
-
+            
+            cs2_top_xycorr = (events[f'{peak_type}s2_area']
+                              * events[f'{peak_type}s2_area_fraction_top']
+                              / self.s2_xy_map(s2_positions, map_name=s2_top_map_name))
+            cs2_bottom_xycorr = (events[f'{peak_type}s2_area']
+                                 * (1 - events[f'{peak_type}s2_area_fraction_top'])
+                                 / self.s2_xy_map(s2_positions, map_name=s2_bottom_map_name))
+            
             # For electron lifetime corrections to the S2s,
             # use drift time computed using the main S1.
             el_string = peak_type + "s2_interaction_" if peak_type == "alt_" else peak_type
             elife_correction = np.exp(events[f'{el_string}drift_time'] / self.elife)
             result[f"{peak_type}cs2_wo_timecorr"] = (cs2_top_xycorr + cs2_bottom_xycorr) * elife_correction
-            result[f"{peak_type}cs2"] = result[f"{peak_type}cs2_wo_elifecorr"] * elife_correction
-            result[f"{peak_type}cs2_bottom"] = cs2_bottom_wo_elifecorr * elife_correction
+
+
+            for partition, func in regions.items():
+                # partitioned SE and EE
+                cond = func(events[f'{peak_type}s2_x'], events[f'{peak_type}s2_y'])
```

Comment:
Can we call it instead of cond something like `_is _in_region` or `mask_in_region`. Makes it a bit easier to read.

Reply:
changed to `partition_mask`